solr/src/main/java/com/yahoo/ycsb/db/SolrClient.java

```diff
@@ -0,0 +1,327 @@
+      }
+
+      UpdateResponse writeResponse;
+      if (batchMode) {
+        writeResponse = client.add(table, updatedDoc, commitTime);
+      } else {
+        writeResponse = client.add(table, updatedDoc);
+        client.commit(table);
+      }
+      return checkStatus(writeResponse.getStatus());
+    } catch (IOException | SolrServerException e) {
+      e.printStackTrace();
+    }
+    return Status.ERROR;
+  }
+
+  /**
+   * Perform a range scan for a set of records in the database. Each field/value pair from the
+   * result will be stored in a HashMap.
+   *
+   * @param table
+   *          The name of the table
+   * @param startkey
+   *          The record key of the first record to read.
+   * @param recordcount
+   *          The number of records to read
+   * @param fields
+   *          The list of fields to read, or null for all of them
+   * @param result
+   *          A Vector of HashMaps, where each HashMap is a set field/value pairs for one record
+   * @return Zero on success, a non-zero error code on error. See this class's description for a
+   *         discussion of error codes.
+   */
+  @Override
+  public Status scan(String table, String startkey, int recordcount, Set<String> fields,
+      Vector<HashMap<String, ByteIterator>> result) {
+    try {
+      Boolean returnFields = false;
+      String[] fieldList = null;
+      if (fields != null) {
+        returnFields = true;
+        fieldList = fields.toArray(new String[fields.size()]);
+      }
+      SolrQuery query = new SolrQuery();
+      query.setQuery("*:*");
+      query.setParam("fq", "id:[ " + startkey + " TO * ]");
+      if (returnFields) {
+        query.setFields(fieldList);
+      }
+      query.setRows(recordcount);
+      final QueryResponse response = client.query(table, query);
+
+      HashMap<String, ByteIterator> entry;
+
+      for (SolrDocument hit : response.getResults()) {
+        entry = new HashMap<String, ByteIterator>((int) response.getResults().getNumFound());
+        for (String field : hit.getFieldNames()) {
+          entry.put(field, new StringByteIterator(String.valueOf(hit.getFirstValue(field))));
+        }
+        result.add(entry);
+      }
+      return checkStatus(response.getStatus());
+    } catch (IOException | SolrServerException e) {
+      e.printStackTrace();
+    }
+    return Status.ERROR;
+  }
+
+  private Status checkStatus(int status) {
+    Status responseStatus;
+    switch (status) {
+    case 0:
+      responseStatus = Status.OK;
+      break;
+    case 400:
+      responseStatus = Status.BAD_REQUEST;
+      break;
+    case 403:
+      responseStatus = Status.FORBIDDEN;
+      break;
+    case 404:
+      responseStatus = Status.NOT_FOUND;
+      break;
+    case 500:
+      responseStatus = Status.ERROR;
+      break;
+    case 503:
+      responseStatus = Status.SERVICE_UNAVAILABLE;
+      break;
+    default:
+      responseStatus = Status.OK;
```

Comment:
Should the default response status be OK? We should explicitly check for good status and not report a false OK.